.github/workflows/release.yml

```diff
@@ -0,0 +1,263 @@
+name: Release
+
+on:
+  # push:
+  #   branches: [auto-release-workflows]
+  workflow_dispatch:
+
+jobs:
+    
+  release:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      R_REMOTES_NO_ERRORS_FROM_WARNINGS: true
+      RSPM: "https://packagemanager.rstudio.com/cran/__linux__/focal/latest"
+
+    steps:
+      - name: Setup Git repository
+        uses: actions/checkout@v4
+
+      - name: Update Ubuntu packages
+        run: sudo apt-get update
+        
+      - name: Setup R
+        uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: 'release'
+
+      - name: Setup pandoc
+        uses: r-lib/actions/setup-pandoc@v2
+        with:
+            pandoc-version: '2.14.0.3'
+
+      - name: Install rmarkdown
+        run: install.packages("rmarkdown")
+        shell: Rscript {0}
+
+# Get previous version and then create the next version   
+      - name: Application Version
+        id: version
+        uses: paulhatch/semantic-version@v5.0.2
+        with:
+          tag_prefix: "v"
+          version_format: "v${major}.${minor}.${patch}"
+
+# Create version.txt to use when changing the version in the index.md
+      - name: Create version.txt
+        run: echo ${{ steps.version.outputs.version }} > version.txt
+
+# Update version and date in SS330_User_Manual.tex
+      - name: Update version and date in manual
+        run: |
+          version <- readLines("version.txt", encoding = "UTF-8")
+          version <- gsub("v", "", version)
+          manual_tex <- readLines("SS330_User_Manual.tex")
+          manual_tex[grep("Version", manual_tex)] <- gsub("[0-9].[0-9][0-9].[0-9][0-9]", version, manual_tex[grep("Version", manual_tex)])
+          date_line <- manual_tex[grep("date[{]", manual_tex)]
+          todays_month <- format(Sys.Date(), "%B")
+          todays_day <- format(Sys.Date(), "%d")
+          todays_year <- format(Sys.Date(), "%Y")
+          todays_date <- paste0("{", todays_month, " ", todays_day, ", ", todays_year, "}")
+          date_line <- gsub("\\{[^{}]*\\}", todays_date, date_line)
+          manual_tex[grep("date[{]", manual_tex)]<- date_line
+          writeLines(manual_tex, "SS330_User_Manual.tex")
```

Comment:
I suggest changing this to
```
          manual_tex <- readLines("SS330_User_Manual.tex")
          manual_tex[grep("Version", manual_tex)] <- gsub(
            pattern = "[0-9].[0-9][0-9].[0-9][0-9]",
            replacement = version,
            x = manual_tex[grep(pattern = "Version", x = manual_tex)]
          )
          date_line <- grep(pattern = "date\\{", x = manual_tex)
          todays_date <- format(x = as.Date("2023-10-1"), "%B %d, %Y")
          manual_tex[date_line] <- gsub(
            pattern = "\\{[A-Za-z0-9, ]+",
            replacement = paste0("{", todays_date),
            x = manual_tex[date_line]
          )
          writeLines(text = manual_tex, con = "SS330_User_Manual.tex")
```
because you do not need to call format three separate times, you can do it in one call. It is good practice to name your input arguments when there is more than one argument passed. You named an object `date_line` but it was not actually the line number, which is what I thought it was going to be. I also cleaned up some of the regex. Feel free to email me offline if you want help with regular expressions.

Reply:
Implementing your suggestion but instead of using todays_date <- format(x = as.Date("2023-10-1"), "%B %d, %Y"), I wrote it as todays_date <- format(x = Sys.Date(), "%B %d, %Y") so that I don't have to manually change the date. It just pulls the date that the action is run.

Reply:
oh shoot, i was testing to see if you would get a 01 or 1 in the date and forgot to change it back. Sorry about that. I will update the above.